src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs

```diff
@@ -243,6 +243,13 @@ private static void ThrowServerException(HttpResponse response, RequestContext r
 
             if (shouldLogAsError)
             {
+                requestContext.Logger.ErrorPii(
```

Comment:
This code would log if we get a response from AAD. But in a scenario when there's an exception sending a request, like a request timing out, the exception just bubbles up the top. 
Top of my head maybe add a try/catch/throw, [somewhere here](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/8f817842e88dac2988f3914c6f5321190abd6cbf/src/client/Microsoft.Identity.Client/OAuth2/OAuth2Client.cs#L108-L133) and log the same.

Reply:
Added